src/rust/lib_ccxr/src/util/log.rs

```diff
@@ -0,0 +1,592 @@
+            ExitCause::Failure => 1,
+            ExitCause::NoInputFiles => 2,
+            ExitCause::TooManyInputFiles => 3,
+            ExitCause::IncompatibleParameters => 4,
+            ExitCause::UnableToDetermineFileSize => 6,
+            ExitCause::MalformedParameter => 7,
+            ExitCause::ReadError => 8,
+            ExitCause::NoCaptions => 10,
+            ExitCause::WithHelp => 11,
+            ExitCause::NotClassified => 300,
+            ExitCause::ErrorInCapitalizationFile => 501,
+            ExitCause::BufferFull => 502,
+            ExitCause::MissingAsfHeader => 1001,
+            ExitCause::MissingRcwtHeader => 1002,
+
+            ExitCause::FileCreationFailed => 5,
+            ExitCause::Unsupported => 9,
+            ExitCause::NotEnoughMemory => 500,
+            ExitCause::Bug => 1000,
+        }
+    }
+}
+
+impl<'a> CCExtractorLogger {
+    /// Returns a new instance of CCExtractorLogger with the provided settings.
+    ///
+    /// `gui_mode` is used to determine if the log massages are intercepted by a GUI.
+    /// `target` specifies the location for printing the log messages.
+    /// `debug_mask` is used to filter debug messages based on its type.
+    pub const fn new(
+        target: OutputTarget,
+        debug_mask: DebugMessageMask,
+        gui_mode: bool,
+    ) -> CCExtractorLogger {
+        CCExtractorLogger {
+            target,
+            debug_mask,
+            gui_mode,
+        }
+    }
+
+    /// Set the mode to Normal or Debug Mode based on `false` or `true` respectively for the
+    /// underlying [`DebugMessageMask`].
+    ///
+    /// This method switches the mask used for filtering debug messages.
+    /// Similar to [`DebugMessageMask::set_debug_mode`].
+    pub fn set_debug_mode(&mut self, mode: bool) {
+        self.debug_mask.set_debug_mode(mode)
+    }
+
+    /// Check if the current mode is set to Debug Mode.
+    ///
+    /// Similar to [`DebugMessageMask::is_debug_mode`].
+    pub fn is_debug_mode(&self) -> bool {
+        self.debug_mask.is_debug_mode()
+    }
+
+    /// Returns the currently set target for logging messages.
+    pub fn target(&self) -> OutputTarget {
+        self.target
+    }
+
+    /// Check if the messages are intercepted by GUI.
+    pub fn is_gui_mode(&self) -> bool {
+        self.gui_mode
+    }
+
+    fn print(&self, args: &Arguments<'a>) {
+        match &self.target {
+            OutputTarget::Stdout => print!("{}", args),
+            OutputTarget::Stderr => eprint!("{}", args),
+            OutputTarget::Quiet => {}
+        }
+    }
+
+    /// Log a fatal error message. Use [`fatal!`] instead.
+    ///
+    /// Used for logging errors dangerous enough to crash the program instantly.
+    pub fn log_fatal(&self, exit_cause: ExitCause, args: &Arguments<'a>) -> ! {
+        self.log_error(args);
+        info!("Issues? Open a ticket here");
+        info!("https://github.com/CCExtractor/ccextractor/issues");
```

Comment:
Both the text in `info!` should end with `\n` since `info!` does not put a newline. (missed this the last review)

Reply:
Done